AISKU/src/AISku.ts

```diff
@@ -200,8 +200,62 @@ export class AppInsightsSku implements IApplicationInsights {
 
             // Will get recalled if any referenced values are changed
             _addUnloadHook(onConfigChange(cfgHandler, () => {
-                if (_config.connectionString) {
-                    const cs = parseConnectionString(_config.connectionString);
+                let configCs =  _config.connectionString;
+
+                function _parseCs() {
+                    return createAsyncPromise<ConnectionString>((resolve, reject) => {
+                        doAwaitResponse(configCs, (res) => {
+                            let curCs = res && res.value;
+                            let parsedCs = null;
+                            if (!res.rejected && curCs) {
+                                // replace cs with resolved values in case of circular promises
+                                _config.connectionString = curCs;
+                                parsedCs = parseConnectionString(curCs);
+                            }
+                            // if can't resolve cs promise, null will be returned
+                            resolve(parsedCs);
+                        });
+                    });
+
+                }
+                
+                if (isPromiseLike(configCs)) {
+                    let ikeyPromise = createAsyncPromise<string>((resolve, reject) => {
+                        _parseCs().then((cs) => {
+                            let ikey = _config.instrumentationKey;
+                            ikey = cs && cs.instrumentationkey || ikey;
+                            resolve(ikey);
+                        }).catch((e) => {
+                            // parseCs will always resolve(unless timeout)
+                            // return null in case any error happens
+                            resolve(null);
+                        });
+
+                    });
+
+                    let urlPromise = createAsyncPromise<string>((resolve, reject) => {
+                        _parseCs().then((cs) => {
+                            let url = _config.endpointUrl;
+                            let ingest = cs && cs.ingestionendpoint;
+                            url = ingest? ingest + DEFAULT_BREEZE_PATH : url;
+                            resolve(url);
```

Comment:
This should also return the `_config.userOverrideEndpointUrl` as if someone passes this and a promise connection string  this would end up overriding their override with the connection string.

Actually, if they provide a user override, we probably just should worry about parsing the connection string here. So avoid all of the lines from 236-248